clang/lib/APINotes/APINotesReader.cpp

```diff
@@ -322,6 +322,44 @@ class FieldTableInfo
   }
 };
 
+BoundsSafetyInfo::BoundsSafetyKind readKindFlag(uint8_t kind) {
+  switch (kind) {
+  case 0x00:
```

Comment:
Alternatively, we could make sure the enum constants have the corresponding values and we could cast to/from the underlying type (and maybe assert that the value is between the expected range).